sdk/eventhub/Azure.Messaging.EventHubs/src/Core/ChannelEnumerableSubscription.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A read-only subscription to a channel, exposed as an
+    ///   asynchronous enumerator.
+    /// </summary>
+    ///
+    /// <typeparam name="T">The type of data contained in the channel.</typeparam>
+    ///
+    /// <seealso cref="IAsyncEnumerable{T}" />
+    ///
+    internal class ChannelEnumerableSubscription<T> : IAsyncEnumerable<T>, IAsyncDisposable
+    {
+        /// <summary>The seed to use for initializing random number generated for a given thread-specific instance.</summary>
+        private static int s_randomSeed = Environment.TickCount;
```

Comment:
`private static readonly`? 🤔

Reply:
It gets updated when the random is created per-thread.   That said, after looking over some feedback, it's going away for now.